content/deployment.md

```diff
@@ -0,0 +1,296 @@
+3. Fix any bugs found in step 2. and repeat.
+4. Once you are satisfied with the staging release, release the *exact same* version to production.
+5. Run final QA on production.
+
+Steps 2. and 5. can be quite time-consuming, especially if you are aiming to maintain a high level of quality in your application. That's why it's a great idea to develop a suite of acceptance tests (see our {% link_to 'testing' 'Testing Article'} for more on this). To take things even further, you could run a load/stress test against your staging server on every release.
+
+<h3 id="rolling-updates-and-data">Rolling deployments and data versions</h3>
+
+It's important to understand what happens during a deployment, especially if your deployment involves changes in data format (and potentially data migrations, see the {% link_to 'collections' 'Collections Article'}).
+
+Depending on where your app is deployed and the number of application processes you have running, things will be different, but if you are deployed in a scaled way to Galaxy, there'll be a period where a number of containers are running the old version, and a number the new, as users are migrated smoothly across to the new version of your app.
+
+[ss]
+
+If the new version involves a different type of data, then you need to be a little more careful about how you step through versions to ensure that all the versions that are deployed simultaneously at all times. You can read more about how to do this in the [collections article](collections.html#migrating).
+
+<h2 id="analytics">Monitoring users via analytics</h2>
+
+It's common to want to know which pages of your app are most commonly visited, and where users are coming from. Here's a simple setup that will get you URL tracking using Google Analytics. We'll be using the [`okgrow:analytics`](https://atmospherejs.com/okgrow/analytics) package.
+
+```
+meteor add okgrow:analytics
+```
+Now, we need to configure the package with our Google Analytics key (the package also supports a large variety of other providers, check out the [documentation on Atmosphere](https://atmospherejs.com/okgrow/analytics)). Pass it in as part of [_Meteor settings_](#environment):
+
+```js
+{
+  "public": {
+    "analyticsSettings": {
+      // Add your analytics tracking id's here
+      "Google Analytics" : {"trackingId": "Your tracking ID"}
+    }
+  }
+}
+```
+
+The analytics package hooks into Flow Router (see the [routing article](routing) for more) and records all of the page events for you.
+
+You may want to track non-page change related events (for instance publication subscription, or method calls) also. To do so you can use the custom event tracking functionality:
+
+```
+Todos.methods.updateText = new ValidatedMethod({
+  ...
+  run({ todoId, newText }) {
+    // We use `isClient` here because we don't want to run it on server-server method simulation
+    if (Meteor.isClient) {
+      analytics.track('Todos.methods.updateText', {todo, newText});
```

Comment:
That's the idea 😉
On Tue, 15 Dec 2015 at 7:41 PM, Sashko Stubailo <notifications@github.com>
wrote:

> In content/deployment.md
> <https://github.com/meteor/guide/pull/109#discussion_r47609482>:
>
> > +    }
> > +  }
> > +}
> > +```
> > +
> > +The analytics package hooks into Flow Router (see the [routing article](routing) for more) and records all of the page events for you.
> > +
> > +You may want to track non-page change related events (for instance publication subscription, or method calls) also. To do so you can use the custom event tracking functionality:
> > +
> > +```
> > +Todos.methods.updateText = new ValidatedMethod({
> > +  ...
> > +  run({ todoId, newText }) {
> > +    // We use `isClient` here because we don't want to run it on server-server method simulation
> > +    if (Meteor.isClient) {
> > +      analytics.track('Todos.methods.updateText', {todo, newText});
>
> Looks nice! Makes me think maybe the package should do it for you.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/meteor/guide/pull/109/files#r47609482>.
>